eos/maths/gegenbauer-polynomial.hh

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ * Copyright (c) 2022 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH
+#define EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH 1
+
+#include <eos/maths/power-of.hh>
+
+#include <vector>
+
+namespace eos
+{
+    class GegenbauerPolynomial
+    {
+        private:
+            const unsigned int order;
+
+            const double alpha; // alpha > -0.5
+
+            std::vector<double> coefficients;
+
+            const unsigned int halforder;
+
+            std::vector<double> calculate_coefficients() const
+            {
+                const unsigned int isodd = order % 2;
+                const double epsilon = 1e-8;
+                std::vector<double> result;
+
+                for (unsigned i = 0 ; i <= halforder ; i++)
+                {
+                    result[i] = power_of<2*i + isodd>(2) * power_of<halforder - i>(-1.0) * 
+                        tgamma(alpha + halforder + i + isodd) / (tgamma(halforder - i + 1) * tgamma(2*i + isodd + 1)); // denominator: (halforder - i)! * (2i + isodd)!
+                }
+
+                if (std::abs(alpha) <= epsilon)
+                    return result;
+                else
+                    for (unsigned i = 0; i <= halforder ; i++)
+                    {
+                        result[i] = result[i] / tgamma(alpha);
+                    }
+
+                    return result;
+            }
+
+        public:
+            GegenbauerPolynomial(const unsigned int order, const double alpha) :
+            order(order),
+            alpha(alpha),
+            halforder(floor(static_cast<double>(order) / 2.0)),
```

Comment:
Ah, I see. So, half order does not need to be documented as an external input. Only the ``order`` and ``alpha``.